src/components/BoxForTransaction.tsx

```diff
@@ -138,49 +143,93 @@ export const BoxForTransaction = () => {
-            payload: {
-              loadingMessage: "Processing Transaction...",
-            },
-          })
+          await web3.eth
+            .estimateGas(transactionParameters)
+            .then(function (gasEstimate) {
+              transactionParameters.gas = gasEstimate
+            })
 
-          const response = await web3.eth.sendTransaction(transactionParameters)
+          try {
+            dispatch({
+              type: "setState",
+              payload: {
+                loadingMessage: "Processing Transaction...",
+              },
+            })
 
-          dispatch({
-            type: "setState",
-            payload: {
-              loadingMessage: null,
-            },
-          })
+            const response = await web3.eth.sendTransaction(
+              transactionParameters,
+            )
 
-          setIsTransactionDetailsModalOpen(false)
-          setIsTransactionResultModalOpen(true)
-          setTransactionResult({
-            blockHash: cleanHex(`${response.blockHash}`, 66),
-            blockNumber: `${response.blockNumber}`.replace(/[^0-9]/g, ""),
-            transactionHash: cleanHex(`${response.transactionHash}`, 66),
-          })
-          dispatch({
-            type: "setState",
-            payload: {
-              loadedFileContents: null,
-              validatedDeposits: [],
-              previouslyDepositedPubkeys: [
-                ...previouslyDepositedPubkeys,
-                ...allPubkeys,
-              ],
-            },
-          })
+            dispatch({
+              type: "setState",
+              payload: {
+                loadingMessage: null,
+              },
+            })
+
+            setIsTransactionDetailsModalOpen(false)
+            setIsTransactionResultModalOpen(true)
+            setTransactionResult({
+              blockHash: cleanHex(`${response.blockHash}`, 66),
+              blockNumber: `${response.blockNumber}`.replace(/[^0-9]/g, ""),
+              transactionHash: cleanHex(`${response.transactionHash}`, 66),
+            })
+            dispatch({
+              type: "setState",
+              payload: {
+                loadedFileContents: null,
+                validatedDeposits: [],
+                previouslyDepositedPubkeys: [
+                  ...previouslyDepositedPubkeys,
+                  ...allPubkeys,
+                ],
+              },
+            })
+          } catch (error) {
+            dispatch({
+              type: "setState",
+              payload: {
+                loadingMessage: null,
+              },
+            })
+            setIsTransactionDetailsModalOpen(false)
+            const serializedError = JSON.stringify(error)
+            const deserializedError = JSON.parse(serializedError)
+            console.log("send tx test:", deserializedError)
+            console.log(error)
+            if (deserializedError !== null) {
+              if (deserializedError.cause.data.message) {
```

Comment:
I'd do `deserializedError?.cause?.data?.message` here. `deserializedError` isn't `null`, but this next line assumes it's therefore a very specifically-shaped object